azblob/zt_s2s_oauth_test.go

```diff
@@ -0,0 +1,152 @@
+package azblob
+
+import (
+	"encoding/base64"
+	chk "gopkg.in/check.v1"
+	"math/rand"
+	"strings"
+)
+
+func (s *aztestsSuite) TestBlockBlobS2SOAuth(c *chk.C) {
+	bsu := getBSU()
+	ocred, err := getOAuthCredential("")
+	c.Assert(err, chk.IsNil)
+
+	bsu = bsu.WithPipeline(NewPipeline(*ocred, PipelineOptions{})) // get based on the OAuth token
+
+	source := bsu.NewContainerURL(newUUID().String())
+	dest := bsu.NewContainerURL(newUUID().String())
+
+	_, err = source.Create(ctx, nil, PublicAccessNone)
+	c.Assert(err, chk.IsNil)
+	_, err = dest.Create(ctx, nil, PublicAccessNone)
+	c.Assert(err, chk.IsNil)
+
+	sourceBlob := source.NewBlockBlobURL("SourceBlob")
```

Comment:
Great job on the tests! It is possible to reduce lines 11-25 to a single, reusable method?